### ISSUE TITLE
page:restore event

### ISSUE THREAD

**Issue by hgani** [turbolinks/turbolinks#135]
Hi,

We are trying to upgrade from the old Turbolinks but couldn't find the equivalent of the 'page:restore' event. 

Is there any plan to support this event?

Thanks

**Comment by sstephenson** [comment]
There’s no direct equivalent because caching in Turbolinks 5 doesn’t work the same way. What kinds of things are you using the page:restore event for?

**Comment by hgani** [comment]
@sstephenson Thanks for responding.

We are using it for two things:
1. We'd like to refresh our navbar when the user is clicking back so that they always see the most up-to-date state (e.g. notification count).
2. We are using a solution which applies CSS styling to the html tag (as described here http://stackoverflow.com/questions/9997192/twitter-bootstrap-fixed-layout-with-scrollable-sidebar/10013643#10013643), so as the user navigates through the pages (including back), we need to add or remove the styling depending on whether the page needs it.

Thanks

**Comment by hgani** [comment]
@sstephenson 

I managed to remove the needs for distinguishing restore vs application visit by following the principles mentioned in the README, such as using data-turbolinks-permanent and making transformations idempotent.

It was a lot of work and at this point I am still not sure 100% whether I've successfully ported all of our javascript functionality.

I suggest we pass a flag to turbolinks:load event handler to enable developer to detect whether a turbolinks:load event is due to an application or a restore visit. This will greatly help developers transition from legacy turbolinks or non-turbolinks apps which will increase adoption for this promising framework/concept.

We can then remind developers that the use of this flag is discouraged (they should attempt to follow the best practice mentioned in the README) and is only there to help them upgrade their legacy app.

Thanks
